tests/FSharp.SystemTextJson.Tests/Test.Record.fs

```diff
@@ -14,6 +14,16 @@ module NonStruct =
         let actual = JsonSerializer.Deserialize """{"ax":1,"ay":"b"}"""
         Assert.Equal({ ax = 1; ay = "b" }, actual)
 
+    [<Fact>]
+    let ``deserialize empty record with ignore-null-values on`` () =
+        let options =
+            JsonSerializerOptions(DefaultIgnoreCondition = Serialization.JsonIgnoreCondition.WhenWritingNull)
+        try
+            JsonSerializer.Deserialize<A>("{}", options) |> ignore
+        with
+        | :? System.NullReferenceException -> failwith "Unexpected NRE."
+        | ex when ex.Message.Contains("Missing field for record type") -> () // It's expected to fail since the record requires its fields to be initialized.
```

Comment:
```
<@ JsonSerializer.Deserialize<A>("{}", options) |> ignore @> |> 
raisesWith<Exception>(fun e -> <@ e.Message.Contains("Missing field for record type" @>)  ()
```
or something similar

Reply:
This is nicer indeed!

Also I actually use Unquote in most of my projects, I'm not even sure why this one doesn't but I might convert it some day.